TRX/Tron-overview.md

```diff
@@ -183,25 +183,29 @@ try {
 # 9. Calculation of block ID
 Block ID is a combination of block height and the hash of the blockheader’s raw data. To get block ID, first hash the raw data of the blockheader and replace the first 8 bytes of the hash with the blockheight, as the following:
 ```
-private byte[] generateBlockId(long blockNum, byte[] blockHash) { 
- byte[] numBytes = Longs.toByteArray(blockNum); 
- byte[] hash = blockHash; 
- System.arraycopy(numBytes, 0, hash, 0, 8); 
+private byte[] generateBlockId(long blockNum, byte[] blockHash) {
+ byte[] numBytes = Longs.toByteArray(blockNum);
+ byte[] hash = blockHash;
+ System.arraycopy(numBytes, 0, hash, 0, 8);
  return hash;
- }
+}
 ```
 BlockHash is the hash of the raw data of the blockheader, which can be calculated as the following:
 ```
 Sha256Hash.of(this.block.getBlockHeader().getRawData().toByteArray())
 ```
 # 10. Construction and signature of transaction
-There are two ways to construct a transaction:
+Based on your own needs, there are two ways to construct the transaction: invoking API on the full node or filling in every fields of a transaction manually at local.
```

Comment:
This would be better read:

"Based on your own needs there are two methods to constructing a transaction. You can either invoke the gRPC or HTTP API through a full node to build the transaction externally, or create the transaction locally."

Reply:
> two methods to constructing a transaction. 

two methods to construct a transaction. 

> invoke the gRPC or HTTP API

invoke the gRPC / HTTP API (using two `or` could be confused 😄 )